vm/src/hint_processor/builtin_hint_processor/bootloader/simple_bootloader_hints.rs

```diff
@@ -0,0 +1,193 @@
+use num_traits::ToPrimitive;
+use std::collections::HashMap;
+
+use felt::Felt252;
+
+use crate::hint_processor::builtin_hint_processor::bootloader::types::{
+    FactTopology, SimpleBootloaderInput,
+};
+use crate::hint_processor::builtin_hint_processor::bootloader::vars;
+use crate::hint_processor::builtin_hint_processor::hint_utils::{
+    get_integer_from_var_name, get_ptr_from_var_name, insert_value_from_var_name,
+    insert_value_into_ap,
+};
+use crate::hint_processor::hint_processor_definition::HintReference;
+use crate::serde::deserialize_program::ApTracking;
+use crate::types::errors::math_errors::MathError;
+use crate::types::exec_scope::ExecutionScopes;
+use crate::vm::errors::hint_errors::HintError;
+use crate::vm::vm_core::VirtualMachine;
+
+/// Implements
+/// n_tasks = len(simple_bootloader_input.tasks)
+/// memory[ids.output_ptr] = n_tasks
+///
+/// # Task range checks are located right after simple bootloader validation range checks, and
+/// # this is validated later in this function.
+/// ids.task_range_check_ptr = ids.range_check_ptr + ids.BuiltinData.SIZE * n_tasks
+///
+/// # A list of fact_toplogies that instruct how to generate the fact from the program output
+/// # for each task.
+/// fact_topologies = []
+pub fn prepare_task_range_checks(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+) -> Result<(), HintError> {
+    // n_tasks = len(simple_bootloader_input.tasks)
+    let simple_bootloader_input: SimpleBootloaderInput =
+        exec_scopes.get(vars::SIMPLE_BOOTLOADER_INPUT)?;
+    let n_tasks = simple_bootloader_input.tasks.len();
+
+    // memory[ids.output_ptr] = n_tasks
+    let output_ptr = get_ptr_from_var_name("output_ptr", vm, ids_data, ap_tracking)?;
+    vm.insert_value(output_ptr, Felt252::from(n_tasks))?;
+
+    // ids.task_range_check_ptr = ids.range_check_ptr + ids.BuiltinData.SIZE * n_tasks
+    const BUILTIN_DATA_SIZE: usize = 8;
+    let range_check_ptr = get_ptr_from_var_name("range_check_ptr", vm, ids_data, ap_tracking)?;
+    let task_range_check_ptr = (range_check_ptr + BUILTIN_DATA_SIZE * n_tasks)?;
+    insert_value_from_var_name(
+        "task_range_check_ptr",
+        task_range_check_ptr,
+        vm,
+        ids_data,
+        ap_tracking,
+    )?;
+
+    // fact_topologies = []
+    let fact_topologies = Vec::<FactTopology>::new();
+    exec_scopes.insert_value(vars::FACT_TOPOLOGIES, fact_topologies);
+
+    Ok(())
+}
+
+/// Implements
+/// %{ ids.num // 2 %}
+pub fn divide_num_by_2(
+    vm: &mut VirtualMachine,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+) -> Result<(), HintError> {
+    let felt = get_integer_from_var_name("num", vm, ids_data, ap_tracking)?.into_owned();
+
+    let num = felt
+        .to_u64()
```

Comment:
Is it clear that `u64` is sufficient? I don't think it is if it comes from here: https://github.com/lambdaclass/cairo-lang/blob/02d511208cece64da1300445ab0586cf75205e70/src/starkware/cairo/bootloaders/simple_bootloader/run_simple_bootloader.cairo#L121

If so, I think a `U256` is appropriate.

Reply:
Yeah you're right, I missed the context of where this is used.